### PR TITLE
Remove `bit_iterator` macro

### LINKED ISSUE #632: Remove `bit_iterator` macro
**Describe what you want implemented** 
Remove the `bit_iterator` macro.

**Describe "Why" this is needed**
As far as I can see we only invoke the macro once to create `BitIterator8` so I don't see the need to have a macro for it. Creating `BitIterator8` directly is much clearer.

**Describe alternatives you've considered**
Leave as is.

**Additional context**
NA

### PR DESCRIPTION
**Pull request by ashutoshvarma**
Fix #632

### CODE REVIEW

**Comment by ashutoshvarma** [comment]
Seems one of the CI checks failed due to an error in uploading a coverage report to codecov.

File: src/bit_iterator.rs [hunk 1]
```diff
@@ -8,54 +8,47 @@
...
-            scalar: E,
-            // num_of_total_bits represents the sum of all of the bits of each
-            // integer If we have 2 u32s then the total number of bits will
-            // be 32 * 2 = 64 bits
-            num_of_total_bits: usize,
-            // bit_len represents the bit length of each integer.
-            // If we have a slice of u32s, then bit_len will be 32
-            bit_len: usize,
-        }
+pub struct BitIterator8<E> {
```
Comment (moCello) [inline src/bit_iterator.rs:11]: You didn't derive the `Debug`, `Clone` and `Copy` traits. Can you still include those?

**Review by moCello — CHANGES_REQUESTED** [review]
Very nice, thanks!
Please also include the `Debug`, `Clone` and `Copy` traits as mentioned in the comments.

And it would be great if you could also add a section in the CHANGELOG under 'Unreleased', like:

### Removed

- Remove `bit_iterator` macro [#632]

And link the issue at the bottom of the CHANGELOG

File: src/bit_iterator.rs (hunk 1 above)
Comment (ashutoshvarma) [reply src/bit_iterator.rs:12]: done!

**Review by moCello — APPROVED** [review]
LGTM

**Pull request merged**